sdk/dotnet/Pulumi/Core/InputList.cs

```diff
@@ -114,10 +114,10 @@ internal InputList<T> Clone()
         #region construct from immutable array
 
         public static implicit operator InputList<T>(ImmutableArray<T> values)
-            => values.SelectAsArray(v => (Input<T>)v);
+            => values.EnsureNotDefault().SelectAsArray(v => (Input<T>)v);
 
         public static implicit operator InputList<T>(ImmutableArray<Output<T>> values)
-            => values.SelectAsArray(v => (Input<T>)v);
+            => values.EnsureNotDefault().SelectAsArray(v => (Input<T>)v);
```

Comment:
This is more reasonable but why these functions and not others? There's many functions in this file accepting ImmutableArray<T> and some of them actually accept it via implicit cast to IEnumerable<T>. 

For example:

```
 public static implicit operator InputList<T>(Output<ImmutableArray<T>> values)
            => new InputList<T>(values);
```

Reply:
I think adding safeguard for the ` public static implicit operator InputList<T>(Output<ImmutableArray<T>> values)` is probably reasonable as well?

This would allow the following conversion to work

```csharp
await AssertCanConvert(Output.Create((ImmutableArray<bool>) default));
```

Not sure what to do about the `IEnumerable` overload, since unfortunately you may indeed have `ImmutableArray` via `IEnumerable`

```csharp
  public static implicit operator InputList<T>(Output<IEnumerable<T>> values)
      => values.Apply(a => ImmutableArray.CreateRange(a));
```